feat: add pubsublite.ordering.mode to kafka connector

dpcollins-google opened the pull request:
This is useful for migration cases using the kafka wire protocol.

Also clean up dependency management and upgrade pubsublite and other versions.

dpcollins-google committed:
feat: add pubsublite.ordering.mode to kafka connector

This is useful for migration cases using the kafka wire protocol.

dpcollins-google committed:
feat: add pubsublite.ordering.mode to kafka connector

This is useful for migration cases using the kafka wire protocol.

Also clean up dependency management.

dpcollins-google committed:
feat: add pubsublite.ordering.mode to kafka connector

This is useful for migration cases using the kafka wire protocol.

Also clean up dependency management.

samarthsingal commented on pom.xml:
```diff
@@ -15,54 +15,73 @@
   <parent>
     <groupId>com.google.cloud</groupId>
     <artifactId>google-cloud-shared-config</artifactId>
-    <version>1.5.4</version>
```
Why are we regressing the version here?

dpcollins-google replied:
We're increasing the version.

samarthsingal commented on pom.xml:
```diff
@@ -15,54 +15,73 @@
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
 
-    <kafka.version>3.3.1</kafka.version>
```
Why are we getting rid of version properties? This is the recommended POM structure for GA libraries I was told.

samarthsingal replied:
In general, can you separate out POM changes you are making for this change and the ones that are purely stylistic? Or are all these changes needed as-is?

dpcollins-google replied:
This also isn't the recommended way to define dependency versions for shared dependencies- I'm converting this to follow the best practice.

samarthsingal replied:
Discussed offline, this now uses BOM structure - https://cloud.google.com/java/docs/bom.

dpcollins-google commented on src/main/java/com/google/pubsublite/kafka/sink/PublisherFactoryImpl.java:
```diff
@@ -82,8 +80,7 @@ public Publisher<MessageMetadata> newPublisher(Partition partition) throws ApiEx
                     responseStream -> {
                       ApiCallContext context =
                           getCallContext(
-                              PubsubContext.of(Framework.of(PUBSUBLITE_KAFKA_SINK_CONNECTOR_NAME)),
```
This is interpreted by the backend and converted to an enum. I need to add it to the enum, and so I'm making it have the same structure as the rest of them.

dpcollins-google commented on src/main/java/com/google/pubsublite/kafka/sink/OrderingMode.java:
```diff
@@ -0,0 +1,21 @@
+package com.google.pubsublite.kafka.sink;
+
+public enum OrderingMode {
+  DEFAULT,
```
Done.

dpcollins-google committed:
feat: add pubsublite.ordering.mode to kafka connector

This is useful for migration cases using the kafka wire protocol.

Also clean up dependency management.

dpcollins-google committed:
feat: add pubsublite.ordering.mode to kafka connector

This is useful for migration cases using the kafka wire protocol.

Also clean up dependency management.

dpcollins-google committed:
feat: add pubsublite.ordering.mode to kafka connector

This is useful for migration cases using the kafka wire protocol.

Also clean up dependency management.
